genericjpa/README.md

```diff
@@ -0,0 +1,84 @@
+Hibernate-Search-JPA
+====================
```

Comment:
I am a bit confused, I wouldn't expect this entire module ("genericjpa") as part of this PR, given you say it's the async backend only. Maybe the split between the two things is not entirely clear to me. What is it you consider as "async backend" and how's that different from "genericjpa"?